src/components/settings-drawer.tsx

```diff
@@ -21,20 +21,40 @@ export interface SettingsDrawerProps extends DrawerProps {
 export default function SettingsDrawer(props: SettingsDrawerProps): React.ReactElement {
   const classes = useStyles();
   const { settings, onSettingsChange, ...otherProps } = props;
-  const { trajectoryAnimation } = settings;
+  const { trajectoryAnimation, trajectoryDiameter, trajectoryColor } = settings;
 
   const trajAnimsText = React.useMemo(
     () => Object.keys(TrajectoryAnimation).slice(Object.keys(TrajectoryAnimation).length * 0.5),
     [],
   );
 
-  const drawerAnchor = useMediaQuery('(max-aspect-ratio: 8/10)') ? 'bottom' : 'right';
+  const trajDiameterText = React.useMemo(
+    () => Object.keys(TrajectoryDiameter).slice(Object.keys(TrajectoryDiameter).length * 0.5),
+    [],
+  );
+
+  const trajColorText = React.useMemo(
+    () => Object.keys(TrajectoryColor).slice(Object.keys(TrajectoryColor).length * 0.5),
+    [],
+  );
+
+  const drawerAnchor = useMediaQuery('(max-aspect-ratio: 8/10') ? 'bottom' : 'right';
 
   function handleTrajectoryAnimationChange(ev: React.ChangeEvent<HTMLInputElement>): void {
     const newSettings: Settings = { ...settings, trajectoryAnimation: Number(ev.target.value) };
     onSettingsChange && onSettingsChange(newSettings);
   }
 
+  function handleTrajectoryDiameterChange(ev: React.ChangeEvent<HTMLInputElement>): void {
+    const newSettings: Settings = { ...settings, trajectoryDiameter: Number(ev.target.value) };
+    onSettingsChange && onSettingsChange(newSettings);
+  }
+
+  function handleTrajectorColorChange(ev: React.ChangeEvent<HTMLInputElement>): void {
```

Comment:
Some tests for this would be nice. You can use `jest.fn()` to provide a mock, simulate a click in enzyme, and use `toHaveBeenCalledTimes(1)` to check if it works.